kse/src/org/kse/gui/dialogs/DViewDHParameters.java

```diff
@@ -0,0 +1,210 @@
+package org.kse.gui.dialogs;
+
+import java.awt.Container;
+import java.awt.Dialog;
+import java.awt.Font;
+import java.awt.Toolkit;
+import java.awt.datatransfer.Clipboard;
+import java.awt.datatransfer.StringSelection;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.text.MessageFormat;
+import java.util.ResourceBundle;
+
+import javax.swing.JButton;
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.JOptionPane;
+import javax.swing.JTextArea;
+import javax.swing.SwingUtilities;
+import javax.swing.filechooser.FileNameExtensionFilter;
+
+import org.bouncycastle.util.io.pem.PemObject;
+import org.bouncycastle.util.io.pem.PemWriter;
+import org.kse.ApplicationSettings;
+import org.kse.crypto.CryptoException;
+import org.kse.gui.CurrentDirectory;
+import org.kse.gui.CursorUtil;
+import org.kse.gui.JEscDialog;
+import org.kse.gui.JavaFXFileChooser;
+import org.kse.gui.LnfUtil;
+import org.kse.gui.PlatformUtil;
+import org.kse.utilities.DialogViewer;
+import org.kse.utilities.pem.PemInfo;
+import org.kse.utilities.pem.PemUtil;
+
+import net.miginfocom.swing.MigLayout;
+
+public class DViewDHParameters extends JEscDialog {
+	
+
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = -1711814777923997727L;
+
+	private static ResourceBundle res = ResourceBundle.getBundle("org/kse/gui/dialogs/resources");
+	
+	private JTextArea jtAreaPem;
+	private JButton jbExport;
+	private JButton jbOK;
+	private JButton jbCopy;
+	private static int keySize;
+	private byte [] dhParameters;
+	private static final String FILE_SUFFIX = ".pem";
+	private static final String EB = "DH PARAMETERS"; //Encapsulation Boundary
+	
+	
+	public DViewDHParameters (JFrame parent, String title, byte[] dhParams)
+		throws CryptoException {
+
+		super(parent, title, Dialog.ModalityType.DOCUMENT_MODAL);
+		this.dhParameters = dhParams;
+		initComponents(dhParameters);
+	}
+	
+	private void initComponents(byte[] dhParams) throws CryptoException {
+		//TODO Generate DH Parameters icon
+	    PemInfo pemInfo = new PemInfo(EB, null, dhParams);
```

Comment:
Indentation seems wrong here

Reply:
It appears correctly on my end, hopefully the new commit will correct it? Let me know please.

Reply:
There are some lines in your code indented with a mix of tabs and spaces, for example this line (85). You don't see any difference because your editor or IDE shows one tab as 4 spaces, but GitHub shows one tab as 8 spaces here and therefore the indentation is wrong, which makes the code hard to read. Most editors/IDEs have an option to display the whitespace:

![grafik](https://user-images.githubusercontent.com/7840192/121781257-33427300-cba4-11eb-9710-8c1ef4d2bc9e.png)

Reply:
Updated, enabled the show whitespace characters and removed the spaces.